### DISCUSSION TITLE
Add more than an image as an input

### DISCUSSION THREAD

**Discussion by RobinFrcd** [ultralytics/ultralytics#4784] [Q&A]
Hi ! I'd like to fine-tune a v8 detection model. Is it possible to add more input than a just the image ?
For example, I'd like to have an image and a class as input. So the network learns that some output classes can only be valid if the input image is of a specific input class.
A solution would be to have a model per input class, but I'd like to have a single big model.

Thanks !

**Comment by glenn-jocher** [comment]
@RobinFrcd hello! Great to hear you're interested in fine-tuning a YOLOv8 model. Currently, YOLOv8 is designed to take images as input for training and inference. The architecture doesn't natively support additional inputs like a separate class label alongside the image.

However, you can achieve a similar effect by incorporating the class information into the image itself, such as overlaying text on the image or modifying the image border color to represent different classes. This way, the model can learn to associate certain visual cues with the desired output classes.

For a more sophisticated approach, you would need to modify the model architecture to accept additional inputs, which would require a custom implementation. This is beyond the scope of the standard YOLOv8 usage and would require significant changes to the codebase.

If you're looking for more advanced customization, you might consider exploring the Enterprise License, which offers more flexibility for commercial product development and allows for modifications like the one you're suggesting.

For further guidance on YOLOv8's capabilities and usage, please refer to our documentation at https://docs.ultralytics.com. 😊
